backend/src/neuroagent/app/routers/threads.py

```diff
@@ -200,18 +212,24 @@ async def get_thread_messages(
     _: Annotated[Threads, Depends(get_thread)],  # to check if thread exists
     thread_id: str,
     tool_list: Annotated[list[type[BaseTool]], Depends(get_tool_list)],
+    entity: list[Literal["USER", "AI_TOOL", "TOOL", "AI_MESSAGE"]] | None = Query(
+        default=None
+    ),
+    sort: Literal["creation_date", "-creation_date"] = "creation_date",
 ) -> list[MessageResponse]:
     """Get all messages of the thread."""
     # Create mapping of tool names to their HIL requirement
     tool_hil_mapping = {tool.name: tool.hil for tool in tool_list}
 
+    if entity:
+        entity_where = or_(*[Messages.entity == ent for ent in entity])
+    else:
+        entity_where = true()
     messages_result = await session.execute(
         select(Messages)
-        .where(
-            Messages.thread_id == thread_id,
-        )
+        .where(Messages.thread_id == thread_id, entity_where)
         .options(joinedload(Messages.tool_calls))  # Eager load tool_calls
-        .order_by(Messages.order)
+        .order_by(desc(Messages.order) if sort.startswith("-") else Messages.order)
```

Comment:
I mean we might as well directly sort on the creation date no? Given the fact we will get rid of the `order` attribute.
As discussed, (unless we missed something) the two approaches should lead to identical sorting.

Or am I missing something. Anyway, we might also do it in a different PR. It is just a bit misleading to pass a query parameter `sort=creation_date` and then sort on a different column